tests/integ_tests/queries_tests.rs

```diff
@@ -0,0 +1,101 @@
+use assert_matches::assert_matches;
+use std::time::Duration;
+use temporal_sdk_core::protos::coresdk::{
+    workflow_activation::{wf_activation_job, WfActivationJob},
+    workflow_commands::{CompleteWorkflowExecution, QueryResult, QuerySuccess, StartTimer},
+    workflow_completion::WfActivationCompletion,
+};
+use temporal_sdk_core::protos::temporal::api::query::v1::WorkflowQuery;
+use temporal_sdk_core::tracing_init;
+use test_utils::{init_core_and_create_wf, with_gw, GwApi};
+
+#[tokio::test]
+async fn simple_query_legacy() {
+    tracing_init();
+    let workflow_id = "simple_query";
+    let query_resp = b"response";
+    let (core, task_q) = init_core_and_create_wf(workflow_id).await;
+    let task = core.poll_workflow_task(&task_q).await.unwrap();
+    core.complete_workflow_task(WfActivationCompletion::from_cmds(
+        vec![
+            StartTimer {
+                timer_id: "timer-1".to_string(),
+                start_to_fire_timeout: Some(Duration::from_millis(500).into()),
+            }
+            .into(),
+            StartTimer {
+                timer_id: "timer-2".to_string(),
+                start_to_fire_timeout: Some(Duration::from_secs(3).into()),
+            }
+            .into(),
+        ],
+        task.run_id.clone(),
+    ))
+    .await
+    .unwrap();
+    tokio::time::sleep(Duration::from_secs(1)).await;
+    // Query after timer should have fired and there should be new WFT
+    let query_fut = with_gw(core.as_ref(), |gw: GwApi| async move {
+        gw.query_workflow_execution(
+            workflow_id.to_string(),
+            task.run_id.to_string(),
+            WorkflowQuery {
+                query_type: "myquery".to_string(),
+                query_args: Some(b"hi".into()),
+            },
+        )
+        .await
+        .unwrap()
+    });
+    let workflow_completions_future = async {
+        // Give query a beat to get going
+        tokio::time::sleep(Duration::from_millis(400)).await;
```

Comment:
I think you know what I have to say about this :)

Reply:
Heh, yeah. Unfortunately there is just no reasonable way to hook into "the query request has been issued" because the query call stays open until fulfilled. I would have to somehow instrument all the way down inside of the http handler and it'd just be a monster mess